metakernel/magics/python_magic.py

```diff
@@ -56,7 +56,7 @@ def input(*args, **kwargs):
                 self.env["__builtins__"]["raw_input"] = self.kernel.raw_input = self.kernel.raw_input
                 self.env["raw_input"] = self.kernel.raw_input = self.kernel.raw_input
         try:
-            return eval(code.strip(), self.env)
+            return eval(code.strip(), self.env, dict())
```

Comment:
Ah, from the docs "If the locals dictionary is omitted it defaults to the globals dictionary".  I'l just remove it.